### ISSUE TITLE
SupportsRound[int] hinted functions work with floats if round is called with no additional args

### ISSUE THREAD

**Issue by leaprovenzano** [python/mypy#8213]
>Note: if you are reporting a wrong signature of a function or a class in
the standard library, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues

this genuinely  might be a [typeshed]( https://github.com/python/typeshed) issue i'm not 100% sure. To be sure some similar issues about `__round__` have cropped up both here and in typeshed:

#6211
python/typeshed#2638

If maintainers think this is solely a typeshed issue and would like me to repost over there please just let me know. If it is a typeshed issue and is a regression ( which it may be)  then perhaps I could help by adding a test somewhere? 

 ### Are you reporting a bug, or opening a feature request?

possible bug

 ###  Example:

```python
import typing

def round_it(x: typing.SupportsRound[int]) -> int:
    return round(x)

round_it(3.2)
```

### What is the actual behavior/output?

though `round` is called with no `ndigits` arg in the example above we still get the following error when running mypy: 

```
scratch.py:11: error: Argument 1 to "round_it" has incompatible type "float"; expected "SupportsRound[int]"
scratch.py:11: note: Following member(s) of "float" have conflicts:
scratch.py:11: note:     Expected:
scratch.py:11: note:         @overload
scratch.py:11: note:         def __round__(self) -> int
scratch.py:11: note:         @overload
scratch.py:11: note:         def __round__(self, ndigits: int) -> int
scratch.py:11: note:     Got:
scratch.py:11: note:         @overload
scratch.py:11: note:         def __round__(self, ndigits: None = ...) -> int
scratch.py:11: note:         @overload
scratch.py:11: note:         def __round__(self, ndigits: int) -> float
```
note that even if I modify [this bit of typeshed](https://github.com/python/typeshed/blob/21a9e696122bafd17e25393296e3920ddb7ed96c/stdlib/2and3/builtins.pyi#L257-L261)  to include : 
```python
@overload
def __round__(self, ndigits: None = ...) -> int: ...
```
I still have the same issue ( just with `<1 more overload not shown>`, etc)

### What is the behavior/output you expect?

since `SupportsRound` is parameterised by return type and I have called the `round` method with no extra args I would expect that the example provided above would pass.  

Specifically : 
given both `float` and `SupportsRound[int]` more or less share a signature for the ` def __round__(self)`  and `round` is called w/ no args.  I would expect mypy to be able to interpret that we're using `def __round__(self, ndigits: None = ...) -> int` and not care about the conflicting overload since it's not called.

### What are the versions of mypy and Python you are using?

python: 3.8.0
mypy: 0.761 (and master)

### What are the mypy flags you are using?

I have tried running with each of the following flag sets :
- no flags
- `--strict`
- `--no-implicit-optional`
- `--no-strict-optional`
- `--strict-optional`

**Comment by hauntsaninja** [comment]
hello! one thing to note is that mypy's issue is nothing to do with line 4 (as running mypy on the following snippet will indicate). mypy is simply concerned whether a `float` is a valid thing to pass to a function that wants a `SupportsRound[int]`.
```
import typing

def round_it(x: typing.SupportsRound[int]) -> int:
    # return round(x)
    pass

round_it(3.2)
```

this is why it's attempting to ensure that `float` satisfies all the necessary overrides that `SupportsRound[int]` demands.
https://github.com/python/mypy/blob/9101707bd0c96624d09cb31fe573d7e25c89a35c/mypy/subtypes.py#L402
mypy looks at `SupportsRound[int]`, sees that it needs something to satisfy the use case of `def (ndigits: builtins.int) -> builtins.int*` but all `float` has to offer is `def (ndigits: builtins.int) -> builtins.float` (and `def (ndigits: None =) -> builtins.int`) and so determines that `float` doesn't satisfy the protocol.

okay, so what now? given that you don't care about the `ndigits` parameter, you get nothing from specifying the generic parameter `_T_co`, in your case, `int`. your use case is fully captured by the first overload that always returns an `int` (side note: maybe `, ndigits: None = ...` should be added to this overload?). https://github.com/python/typeshed/blob/f5a1925e765b92dd1b12ae10cf8bff21c225648f/stdlib/3/typing.pyi#L115

so i'd just use:
```
import typing

def round_it(x: typing.SupportsRound[typing.Any]) -> int:
    return round(x)

round_it(3.2)
```
or omit the type parameter entirely, if not using `--strict`.

note this isn't a shady cover-up, because of the first overload, mypy knows we're getting an `int`.
```
def round_it(x: typing.SupportsRound[typing.Any]) -> None:
    reveal_type(round(x))
```

if you do need to use `ndigits` and run into issues, the solution is probably to use overloads, as `round` itself does:
https://github.com/python/typeshed/blob/f5a1925e765b92dd1b12ae10cf8bff21c225648f/stdlib/2and3/builtins.pyi#L1386

disclaimer: i'm new to mypy so anything i say might be totally off the mark :-)
